exercises/src/main/scala/fpinscala/monads/Monad.scala

```diff
@@ -34,25 +34,49 @@ trait Monad[M[_]] extends Functor[M] {
   def map2[A,B,C](ma: M[A], mb: M[B])(f: (A, B) => C): M[C] =
     flatMap(ma)(a => map(mb)(b => f(a, b)))
 
-  def sequence[A](lma: List[M[A]]): M[List[A]] = ???
-
-  def traverse[A,B](la: List[A])(f: A => M[B]): M[List[B]] = ???
-
-  def replicateM[A](n: Int, ma: M[A]): M[List[A]] = ???
-
-  def compose[A,B,C](f: A => M[B], g: B => M[C]): A => M[C] = ???
+  def sequence[A](lma: List[M[A]]): M[List[A]] =
+    lma.foldRight(unit(Nil:List[A]))((p, acc) => {
+      map2(p, acc)(_ :: _)
+    })
```

Comment:
Also only about style, but you can use `{}` instead of `()` to pass a function argument, so that you don't have to add `{}` inside for multi-line blocks. e.g.

```
  lma.foldRight(unit(List.empty[A])) { (p, acc) => 
    map2(p, acc)(_ :: _)
  }
```